jormungandr-lib/src/interfaces/fragment_log_persistent.rs

```diff
@@ -14,3 +28,88 @@ pub struct PersistentFragmentLog {
     #[serde(with = "FragmentDef")]
     pub fragment: Fragment,
 }
+
+pub struct FileFragments {
+    reader: BufReader<fs::File>,
+    file_path: PathBuf,
+}
+
+pub struct FileFragmentsIterator {
+    reader: BufReader<fs::File>,
+    file_path: PathBuf,
+    counter: usize,
+}
+
+impl FileFragments {
+    pub fn from_path(file_path: PathBuf) -> std::io::Result<Self> {
+        let mut reader = BufReader::new(fs::File::open(file_path.clone())?);
+        reader.fill_buf()?;
```

Comment:
I would rather obtain the file metadata, check its length and return an empty iterator. But that works as a workaround too.

Reply:
Actually, it may be worth using the metadata. We would avoid filling the buffer till it is really needed. Will do.